src/dscanner/analysis/del.d

```diff
@@ -6,30 +6,30 @@
 module dscanner.analysis.del;
 
 import std.stdio;
-import dparse.ast;
-import dparse.lexer;
 import dscanner.analysis.base;
+import dscanner.analysis.helpers;
 import dsymbol.scope_;
 
 /**
  * Checks for use of the deprecated 'delete' keyword
  */
-final class DeleteCheck : BaseAnalyzer
+extern(C++) class DeleteCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	mixin AnalyzerInfo!"delete_check";
 
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName);
 	}
 
-	override void visit(const DeleteExpression d)
+	override void visit(AST.DeleteExp d)
 	{
-		addErrorMessage(d.line, d.column, "dscanner.deprecated.delete_keyword",
+		addErrorMessage(cast(ulong) d.loc.linnum, cast(ulong) d.loc.charnum, "dscanner.deprecated.delete_keyword",
 				"Avoid using the 'delete' keyword.");
-		d.accept(this);
+		// d.accept(this);
```

Comment:
please don't leave commented code in PRs. If that line is needed, we have git history.